### DISCUSSION TITLE
How to derive ChatTheme using ThemeData?

### DISCUSSION THREAD

**Discussion by ppipada** [flyerhq/flutter_chat_ui#616] [Q&A] [answered]
Hi,

I am adding my preferred colors etc in ThemeData in a flutter app. 
Is there a way to use the set colors in theme data in flutter_chat_ui?
As of now, it seems like the widget uses its own colors.

I am a newbie, so even if there is no direct way as of now, any guidance on how to achieve this will be highly appreciated. 

Thanks and regards,

**Comment by demchenkoalex** [comment]
Hey, sorry I was on vacation. You mean flutter_chat_ui should read theme data passed to the entire app?

**Comment by ppipada** [comment]
Thank you for your response, 

On your question: I am not sure if flutter chat ui should read the theme data or should have a option to map themedata to flutter chat ui colors. 

Usecase is: 
- I want to use flutter chat ui as a module inside a larger application. 
- Therefore, I want the look and feel of the chat ui to be same as other parts of the app. 
- For the full app, I am using theme data in flutter that allows me to set colors and other styling params in one place and then other modules/components use that for styling. 
- As of now, I see that for flutter chat ui, it doesnt really read theme data, but has its own styling config. 

Now the usecase can be solved in two ways I think:
1) Either flutter chat ui uses themedata as first preference and any styling input as "overrides". But I suppose, this is very destructive. OR
2) There is a function that takes input as themedata and gives out flutter chat ui styling object, that can be then passed to flutter chat ui initialization.  (side note: Though, I am not sure how this will work out when a flutter app switches themes. e.g: light to dark mode via a switch.)

I am trying to do option 2, but I am not finding any straightforward way to map the styling from theme data to flutter styles. I am not really able to map things cleanly and completely is the issue. Hence the question. 

Any guidance on either options or any easier way to achieve this would be great.

Thanks again,

**Reply by demchenkoalex** [reply]
so currently there is a ChatTheme class where you can provide same values you used for your ThemeData. As I am working on a full rewrite right now I will not be changing that, however your feedback is valuable for v2 and I will make sure to explore how can I feed (or listen to) ThemeData to the chat in a more automatic way.

**Reply by ppipada** [reply]
Thank you for your work !!!

**Comment by demchenkoalex** [comment, marked as answer]
natively supported in v2 https://github.com/flyerhq/flutter_chat_ui/issues/646
